### PR TITLE
Update view command to create symbolic links when the source itself is a symbolic link.

### PR DESCRIPTION
**Pull request by gartung**
Update view command to create symbolic links when the source itself is a symbolic link. This avoids duplicate copied of libraries when the symbolic link is the shortened version of the library name.

### CODE REVIEW

**Comment by mathstuf** [comment]
This is likely going to conflict with #5415.

File: lib/spack/spack/cmd/view.py [hunk 1]
```diff
@@ -241,7 +241,11 @@ def link_one(spec, path, link=os.symlink, verbose=False):
                     continue    # silence these
                 tty.warn("Skipping existing file: %s" % dst)
                 continue
-            link(src, dst)
+            if os.path.islink(src):
+                linkto = os.readlink(src)
```
Comment (mathstuf) [inline lib/spack/spack/cmd/view.py]: Should we resolve the path that is returned here? Or are we trusting that the path it points to will be linked locally as well?

**Comment by gartung** [comment]
OK

**Comment by mathstuf** [comment]
Well, a conflict isn't a reason to close it. That was more of a notification :) .

File: lib/spack/spack/cmd/view.py (hunk 1 above)
Comment (gartung) [reply lib/spack/spack/cmd/view.py]: In practice I ended up with copies of all libraries, even ones that were symbolic links within the install directory. For example, libfoo.so ->libfoo.1.so->libfoo.1.2.3.so would end up with three identical copies of libfoo.1.2.3.so.

Comment (mathstuf) [reply lib/spack/spack/cmd/view.py]: View should never copy anyways. It should hardlink or symlink. What was the link count on the "identical copies"?

Comment (mathstuf) [reply lib/spack/spack/cmd/view.py]: Hmm, they shouldn't be appearing as real files if they're hardlinks to symlinks. Here's what I see:

```
total 0
lrwxrwxrwx. 2 boeckb boeckb 4 Oct 19 15:30 hard-to-symlink -> real
-rw-r-----. 1 boeckb boeckb 0 Oct 19 15:30 real
lrwxrwxrwx. 2 boeckb boeckb 4 Oct 19 15:30 symlink -> real
```

Note the `2` in the second column indicating that there are multiple names for the same file content.

Comment (gartung) [reply lib/spack/spack/cmd/view.py]: It's been so long I don't recall what prompted me to fix this.

**Comment by gartung** [comment]
Will resubmit when I have a concrete example of the bug.

**Pull request closed without merging**